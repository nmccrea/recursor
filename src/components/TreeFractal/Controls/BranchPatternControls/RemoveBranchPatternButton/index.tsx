import React from "react"
import { useDispatch } from "react-redux"
import { removeOne } from "../../../../../state/branchPatterns/actions"
import { BranchPatternId } from "../../../../../state/branchPatterns/types"

interface RemoveBranchPatternButtonProps {
  branchPatternId: BranchPatternId
}

const RemoveBranchPatternButton = ({
  branchPatternId,
}: RemoveBranchPatternButtonProps) => {
  const dispatch = useDispatch()
  return (
    <button
      onClick={() => {
        dispatch(removeOne(branchPatternId))
      }}
    >
      Remove
    </button>
  )
}

export default RemoveBranchPatternButton
