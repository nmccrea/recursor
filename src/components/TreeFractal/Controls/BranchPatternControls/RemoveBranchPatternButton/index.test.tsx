import React from "react"
import * as reactRedux from "react-redux"
import { render, fireEvent, getByText } from "@testing-library/react"
import RemoveBranchPatternButton from "."

describe("`<RemoveBranchPatternButton />`", () => {
  it("renders correctly", () => {
    jest.spyOn(reactRedux, "useDispatch").mockReturnValue(jest.fn())

    const { container } = render(
      <RemoveBranchPatternButton branchPatternId={"test/subject"} />
    )

    expect(container).toMatchSnapshot()
  })

  it("responds correctly to a click event", () => {
    const mockDispatch = jest.fn()
    jest.spyOn(reactRedux, "useDispatch").mockReturnValue(mockDispatch)
    const { container } = render(
      <RemoveBranchPatternButton branchPatternId={"test/subject"} />
    )

    fireEvent.click(getByText(container, "Remove"))

    expect(mockDispatch).toHaveBeenCalledWith({
      type: "branchPatterns/REMOVE_ONE",
      payload: "test/subject",
    })
  })
})
