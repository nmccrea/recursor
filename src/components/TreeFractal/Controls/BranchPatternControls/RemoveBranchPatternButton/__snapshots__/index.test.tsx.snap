// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<RemoveBranchPatternButton />\` renders correctly 1`] = `
<div>
  <button>
    Remove
  </button>
</div>
`;
